olp-cpp-sdk-authentication/tests/CryptoTest.cpp

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (C) 2020 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#include <gtest/gtest.h>
+
+#include <sstream>
+
+#include <olp/authentication/Crypto.h>
+
+namespace {
+
+using olp::authentication::Crypto;
+
+std::string ToString(Crypto::Sha256Digest hash) {
+  std::stringstream stream;
+  for (auto element : hash) {
+    stream << std::hex << std::setw(2) << std::setfill('0')
+           << static_cast<int32_t>(element);
+  }
+  return stream.str();
+}
+
+TEST(CryptoTest, Sha256) {
```

Comment:
Please also add test for HmacSha256 method. There are some generators online available which you can use as a reference to validate output.